test/cql-pytest/conftest.py

```diff
@@ -221,3 +221,24 @@ def temp_workdir():
     """ Creates a temporary work directory, for the scope of a single test. """
     with tempfile.TemporaryDirectory() as workdir:
         yield workdir
+
+# The "tablets" experimental feature is optionally enabled in Scylla, and
+# some tests are expected to fail on Scylla when this option is enabled,
+# but expected to pass without it enabled (and also pass on Cassandra).
+# These test should use the fails_with_tablets fixture.
+# When tablets mode becomes mainstream and no test fails with it, this
+# fixture can be removed.
+@pytest.fixture(scope="session")
+def check_experimental_tablets(cql):
+    # If not running on Scylla, return false.
+    names = [row.table_name for row in cql.execute("SELECT * FROM system_schema.tables WHERE keyspace_name = 'system'")]
+    if not any('scylla' in name for name in names):
+        return False
+    # In Scylla, we check Tablets mode by inspecting the configuration via CQL.
+    experimental_features = list(cql.execute("SELECT value FROM system.config WHERE name = 'experimental_features'"))[0].value
+    return '"tablets"' in experimental_features
```

Comment:
This is not reliable. Soon, tablets will be enabled without any experimental feature. A better way is to check entries in `system_schema.scylla_keyspaces`, see my PR on how this can be done.

Reply:
Maybe. I'll take a look. But this code doesn't need to be future-proof - the implementation can easily be changed later. This is not the important part of my patch - the important part is that I want to check whether the currently-running Scylla uses tablets or not, instead of needing the person or script that ran Scylla to tell us how they ran it.